presto-main/src/main/java/com/facebook/presto/operator/JoinOperatorInfo.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.LookupJoinOperators.JoinType;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Arrays;
+
+import static com.facebook.presto.operator.JoinStatisticsCounter.HISTOGRAM_SIZE;
+import static com.google.common.base.Preconditions.checkState;
+
+public class JoinOperatorInfo
+        implements Mergeable<JoinOperatorInfo>, OperatorInfo
+{
+    private final JoinType joinType;
+    private final long[] logHistogramProbes;
+    private final long[] logHistogramOutput;
+
+    public static JoinOperatorInfo createJoinOperatorInfo(JoinType joinType, long[] logHistogramCounters)
+    {
+        long[] logHistogramProbes = new long[HISTOGRAM_SIZE];
+        long[] logHistogramOutput = new long[HISTOGRAM_SIZE];
+        for (int i = 0; i < HISTOGRAM_SIZE; i++) {
+            logHistogramProbes[i] = logHistogramCounters[2 * i];
+            logHistogramOutput[i] = logHistogramCounters[2 * i + 1];
```

Comment:
Generally I'm all for parenthesis, but there is some level - general knowledge. I think nobody will confuse that * goes before +. Similarly, we do like `a > b && x != y` without parenthesis.